host/sgx/enter.c

```diff
@@ -156,14 +159,39 @@ void oe_enter(
     uint64_t* arg4,
     oe_enclave_t* enclave)
 {
-    // The general purpose registers are preserved by the compiler.
-    // The floating point state and the flags must be explicitly preserved.
-    // The space for saving the floating-point state must be 16 byte aligned.
-    OE_ALIGNED(16)
-    uint64_t fx_state[64];
+    // Additional control registers that need to be preserved as part of the
+    // Windows and Linux x64 ABIs
+    uint32_t mxcsr = 0;
+    uint16_t fcw = 0;
+
     oe_ecall_context_t ecall_context = {{0}};
     _setup_ecall_context(&ecall_context);
 
+    // Compiler will usually handle this on exiting a function that uses AVX,
```

Comment:
Thanks for the nice description. In simulation mode too the jump to the enclave code happens unbeknownst to the compiler. It would be good to refactor this into a separate function and call that in oe_enter_sim as well.

Reply:
Thanks for the catch, fixed.